nasa/earthdata-search
# EDSC-4051, EDSC-4052: Added fetchSwodlrOrder and made sure the download status and results show up on the downloads page

# Overview

### What is the feature?

Completed swodlr integration into EDSC (frontend and backend)

### What is the Solution?

Backend:
- Added submitSwodlrOrder and fetchSwodlrOrder which are lambdas that will submit jobs/orders and retrieve the status of those orders the db from SWODLR.
- Added front end changes so that 1 SWODLR order has 10 granule (split to 10 sub orders) with a max of 10 granules per order request.
- formatted the downloads page to track and display ongoing and completed SWODLR downloads

Frontend:
- Added UI components for SWODLR customization
- Added unit tests for SWODLR components

### What areas of the application does this impact?

- Projects view is updated when a collection is serviceable by SWODLR
- Downloads features SWODLR options and data now

# Testing

### Reproduction steps

- **Environment for testing: prod**
- **Collection to test with: C2799438271-POCLOUD**

There are a few iterations to test: 

Scenario 1: A user selects a few granules to customize and download with SWODLR
- Search for the recommended collection
- Click on it to see the granules
- Select at least one but no more than 9 granules
- Click download or select my project
- Select SWODLR access method
- Customize the granule in any configuration and download
- Open TablePlus or your postgresql db and find the entry in retrieval_orders for the orders you just made (should find empty orders with the same retrieval_collection_id) equal the number of granules you decided to download.
- Create a file called test_swodlr_order.json with the following `{
    "Records": [
      {
        "body": "{\"accessToken\": \"<your prod accessToken>\", \"id\":<retrieval_order.id>}"
      }
    ]
  }`
- Run `npm run invoke-local -- --function submitSwodlrOrder -p <path to test_swodlr_order.json>`
- Go to localhost:8080/downloads and click the first order and observe the status of the downloads
- This will submit the order to SWODLR for that specific granule. Repeat for the other granules.
- Ensure entries have now shown up in https://swodlr.podaac.earthdatacloud.nasa.gov/generatedProductHistory for all the granules.
- After about 5-10 min. The jobs should have started. Create a file called test_fetch_swodlr_order.json and add the following `{
  "accessToken": "<your edl prod accessToken>",
  "id": <retrieval_order.id>,
  "orderType": "SWODLR"
}`
- Run `npm run invoke-local -- --function fetchSwodlrOrder -p <path to test_fetch_swodlr_order.json>`
- Go to localhost:8080/downloads and click the first order and observe the status of the downloads has changed or remained the same and compare this to your orders in https://swodlr.podaac.earthdatacloud.nasa.gov/generatedProductHistory

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.
Downloads page when 1 of 3 orders fails
![image](https://github.com/nasa/earthdata-search/assets/8591291/57f98601-dfdb-46dc-a7c5-daac87a2201e)
<img width="1724" alt="image" src="https://github.com/nasa/earthdata-search/assets/8591291/0a2d8c60-1112-4110-9b77-d079d7479a58">
<img width="1724" alt="Screenshot 2024-06-14 at 4 37 34 PM" src="https://github.com/nasa/earthdata-search/assets/8591291/20f98fa8-4990-4058-b1b5-4c1f87c87f68">
<img width="1724" alt="Screenshot 2024-06-14 at 4 37 46 PM" src="https://github.com/nasa/earthdata-search/assets/8591291/e5105fb5-910b-4958-9d9f-ae6f382e7424">
<img width="1724" alt="Screenshot 2024-06-14 at 5 19 18 PM" src="https://github.com/nasa/earthdata-search/assets/8591291/b1e1e062-b380-40d4-b4fd-34e26820a52a">
<img width="1724" alt="Screenshot 2024-06-14 at 5 19 28 PM" src="https://github.com/nasa/earthdata-search/assets/8591291/4639e873-df85-42b4-af24-cd8fae198bfb">
<img width="1724" alt="image" src="https://github.com/nasa/earthdata-search/assets/8591291/33ba7cec-6d58-4d4b-a6ef-ac5d46eb1d63">
<img width="1724" alt="Screenshot 2024-06-18 at 1 00 08 PM" src="https://github.com/nasa/earthdata-search/assets/8591291/b33457ae-2ecc-4e3b-9f25-d933f8826d86">



# Checklist
